imagetest/test_suites/image_validation/cloud_test.go

```diff
@@ -0,0 +1,90 @@
+package imagevalidation
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+const (
+	chronyService = "chronyd"
+	ntpService    = "ntp"
+	ntpdService   = "ntpd"
+)
+
+var ntpConfig = []string{"/etc/ntp.conf"}
+var chronyConfig = []string{"/etc/chrony.conf", "/etc/chrony/chrony.conf", "/etc/chrony.d/gce.conf"}
+
+// TestNTPService Verify that ntp package exist and configuration is correct.
+// debian 9, ubuntu 16.04 ntp
+// sles-12 ntpd
+// other distros chronyd
+func TestNTPService(t *testing.T) {
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("Couldn't get image from metadata")
+	}
+	var servicename string
+	var configPaths []string
+	switch {
+	case strings.Contains(image, "debian-9"):
+	case strings.Contains(image, "ubuntu-1604"):
+	case strings.Contains(image, "ubuntu-minimal-1604"):
+		servicename = ntpService
+		configPaths = ntpConfig
+	case strings.Contains(image, "sles-12"):
+		servicename = ntpdService
+		configPaths = ntpConfig
+	default:
+		servicename = chronyService
+		configPaths = chronyConfig
+	}
+
+	ntpConfigs, err := readNTPConfig(configPaths)
+	if err != nil {
+		t.Fatalf("failed reading ntp config file %s", err)
+	}
+
+	// The logic here expects at least one 'server' line in /etc/ntp.conf or
+	// /etc/chrony.conf where the first 'server' line points to our metadata
+	// server, but without caring where any subsequent server lines point.
+	// For example, Ubuntu uses metadata.google.internal in the first server line
+	// and ntp.ubuntu.com on the second.
+	for _, config := range ntpConfigs {
```

Comment:
For chrony it may be easier to use `chronyc sources` and look at the output for the metadata server.